### PR TITLE
ToricVarieties: Rename more constructors - Part 2

### PR DESCRIPTION
**Pull request by HereAround**
This PR aims at completing the conversion from `CamelCase` to `snake_case` constructors for the toric varieties. Here, we change the following constructors:
* `ToricDivisor` -> `toric_divsior`,
* `DivisorOfCharacter` -> `divisor_of_character`,
* `ToricDivisorClass` -> `toric_divisor_class`,
* `ToricLineBundle` -> `toric_line_bundle`,
* `ToricMorphism` -> `toric_moprhism`,
* `ToricIdentityMorphism` -> `toric_identity_morphism`.

I also add a constructor `toric_vanishing _set`.

### CODE REVIEW

File: test/ToricVarieties/affine_normal_varieties.jl [hunk 1]
```diff
@@ -1,3 +1,4 @@
+
```
Comment (fingolfin) [inline test/ToricVarieties/affine_normal_varieties.jl]: ?

**Review by fingolfin — APPROVED** [review]
LGTM

File: test/ToricVarieties/affine_normal_varieties.jl (hunk 1 above)
Comment (HereAround) [reply test/ToricVarieties/affine_normal_varieties.jl]: Thx. This was there by accident. I just removed it.

**Comment by fingolfin** [comment]
It seems you did not yet update `test/Serialization/ToricGeometry.jl`

**Comment by fingolfin** [comment]
Also ``src/Serialization/ToricGeometry.jl`

How did you locate the places to change? I would recommend using `git grep` or ripgrep to search through the whole repository.

**Comment by HereAround** [comment]
I used `git grep` and was aware of both files. For instance, `src/Serialization/ToricGeometry.jl` requires (to the best of my understanding) no changes at all. It uses the toric types and creates one instance of the type `ToricDivisor`.

As for `test/Serialization/ToricGeometry.jl` - it must have simply slipped my attention. I have changed it just now. So hopefully, all tests go green in 2 hours from now.

**Comment by HereAround** [comment]
@fingolfin Looks like we are finally there. Sorry this took so long to sort out the tests.

**Pull request merged**